src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,423 @@
+            assert_eq!(reward_addrs.len(), 0);
+        } else if cur_reward_cycle < EXPECTED_FIRST_V2_CYCLE as u128 {
+            // After the start of Alice's first cycle, but before the first V2 cycle,
+            //  Alice is the only Stacker, so check that.
+            let (amount_ustx, pox_addr, lock_period, first_reward_cycle) =
+                get_stacker_info(&mut peer, &key_to_stacks_addr(&alice).into()).unwrap();
+            eprintln!("\nAlice: {} uSTX stacked for {} cycle(s); addr is {:?}; first reward cycle is {}\n", amount_ustx, lock_period, &pox_addr, first_reward_cycle);
+
+            // one reward address, and it's Alice's
+            // either way, there's a single reward address
+            assert_eq!(reward_addrs.len(), 1);
+            assert_eq!(
+                (reward_addrs[0].0).version,
+                AddressHashMode::SerializeP2PKH.to_version_testnet()
+            );
+            assert_eq!((reward_addrs[0].0).bytes, key_to_stacks_addr(&alice).bytes);
+            assert_eq!(reward_addrs[0].1, 1024 * POX_THRESHOLD_STEPS_USTX);
+        } else {
+            // v2 reward cycles have begun, so reward addrs should be read from PoX2 which is Bob + Alice
+            assert_eq!(reward_addrs.len(), 2);
+            assert_eq!(
+                (reward_addrs[0].0).version,
+                AddressHashMode::SerializeP2PKH.to_version_testnet()
+            );
+            assert_eq!((reward_addrs[0].0).bytes, key_to_stacks_addr(&bob).bytes);
+            assert_eq!(reward_addrs[0].1, 512 * POX_THRESHOLD_STEPS_USTX);
+
+            assert_eq!(
+                (reward_addrs[1].0).version,
+                AddressHashMode::SerializeP2PKH.to_version_testnet()
+            );
+            assert_eq!((reward_addrs[1].0).bytes, key_to_stacks_addr(&alice).bytes);
+            assert_eq!(reward_addrs[1].1, 512 * POX_THRESHOLD_STEPS_USTX);
+        }
+    };
+
+    // first tenure is empty
+    peer.tenure_with_txs(&[], &mut coinbase_nonce);
+
+    let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_balance, 1024 * POX_THRESHOLD_STEPS_USTX);
+
+    let alice_account = get_account(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(
+        alice_account.stx_balance.amount_unlocked(),
+        1024 * POX_THRESHOLD_STEPS_USTX
+    );
+    assert_eq!(alice_account.stx_balance.amount_locked(), 0);
+    assert_eq!(alice_account.stx_balance.unlock_height(), 0);
+
+    // next tenure include Alice's lockup
+    let tip =
+        SortitionDB::get_canonical_burn_chain_tip(&peer.sortdb.as_ref().unwrap().conn()).unwrap();
+    let alice_lockup = make_pox_lockup(
+        &alice,
+        0,
+        1024 * POX_THRESHOLD_STEPS_USTX,
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(&alice).bytes,
+        4,
+        tip.block_height,
+    );
+    let tip_index_block = peer.tenure_with_txs(&[alice_lockup], &mut coinbase_nonce);
```

Comment:
looks pretty clean the way that you were able to use "tenure_with_txs".. is it possible to indicate, at least on some more instances, where the "counter" is? It's easier to get lost as teh number of iterations goes up.